test/manipulation.test.js

```diff
@@ -1096,6 +1096,157 @@ describe('manipulation', function() {
       });
     });
   }
+  if (!getSchema().connector.replaceById) {
+    describe.skip('replaceById - not implemented', function() {});
+  } else {
+    describe('replaceOrCreate when forceId is true', function() {
+      var Post;
+      var ds = getSchema();
+      before(function(done) {
+        Post = ds.define('Post', {
+          title: {type: String, length: 255, index: true},
+          content: {type: String},
+          comments: [String],
+        }, {forceId: true});
+        ds.automigrate('Post', done);
+      });
+
+      it('fails without options on create', function(done) {
+        var post = {id: 123, title: 'a', content: 'AAA'};
+        Post.replaceOrCreate(post, function(err, p) {
+          should.exist(err);
+          err.should.match(/'Could not replace. Object with id ' + post.id + ' does not exist!'/);
+          done();
+        });
+      });
+
+      it('fails with options on create', function(done) {
+        var post = {id: 123, title: 'a', content: 'AAA'};
+        Post.replaceOrCreate(post, {validate: false}, function(err, p) {
+          should.exist(err);
+          err.should.match(/'Could not replace. Object with id ' + post.id + ' does not exist!'/);
+          done();
+        });
+      });
+
+      it('works without options on update (promise variant)', function(done) {
+        var post = {title: 'a', content: 'AAA', comments: ['Comment1']};
+        Post.create(post)
+          .then(function(created) {
+            created = created.toObject();
+            delete created.comments;
+            delete created.content;
+            created.title = 'b';
+            return Post.replaceOrCreate(created)
+            .then(function(p) {
+              should.exist(p);
+              p.should.be.instanceOf(Post);
+              p.id.should.equal(created.id);
+              p.should.not.have.property('_id');
+              p.title.should.equal('b');
+              p.should.have.property('content', undefined);
+              p.should.have.property('comments', undefined);
+              return Post.findById(created.id)
+              .then(function(p) {
+                p.should.not.have.property('_id');
+                p.title.should.equal('b');
+                should.not.exist(p.content);
+                should.not.exist(p.comments);
+                done();
+              });
+            });
+          })
+        .catch(done);
+      });
+
+      it('works with options on update (promise variant)', function(done) {
```

Comment:
IMO, there is no need to test all variants of `updateOrCreate`, because `forceId` flag is not affecting them. Pick a single variant and discard the rest. I would personally choose promise API + no options, as it's the simplest one.